### PR TITLE
add reregister command

### PR DESCRIPTION
**Pull request by harshkothari410**

### CODE REVIEW

File: src/commands/index.js [hunk 1]
```diff
@@ -24,24 +24,24 @@ const pj = require('../../package.json');
...
-.command('register --name <name> --url <url> --token <token>',
-  'Register collector with an instance of Refocus')
 .command('start --collectorName <name> --refocusUrl <url> --accessToken <token>',
   'Start collector')
 .command('stop --name <name>', 'Stop collector')
 .command('status --name <name>', 'Show current status of collector')
-.command('deregister --name <name>', 'Deregister collector');
+.command('deregister --name <name>', 'Deregister collector')
+.command('reregister --collectorName <name> --refocusUrl <url> --accessToken <token>',
+  'Re-register collector');
```
Comment (iamigo) [inline src/commands/index.js]: `Reregister a collector which has been deregistered`

File: src/commands/index.js [hunk 2]
```diff
@@ -24,24 +24,24 @@ const pj = require('../../package.json');
...
 
 program.on('--help', () => {
   console.log('  Examples:\n');
   console.log('    $ refocus-collector --help');
-  console.log('    $ refocus-collector register --name=test ' +
+  console.log('    $ refocus-collector start --name=test ' +
```
Comment (iamigo) [inline src/commands/index.js]: is it `--collectorName` (as in lines 27-33) or `--name` (as in lines 38-43)?

File: src/commands/refocus-collector-reregister.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  process.exit(1);
+} else if (!refocusUrl) {
+  logger.error('You must specify the url of the refocus instance');
+  process.exit(1);
+} else if (!accessToken) {
+  logger.error('You must specify an access token');
+  process.exit(1);
+}
+
+console.log('Re-register =>', collectorName, refocusUrl, accessToken);
```
Comment (iamigo) [inline src/commands/refocus-collector-reregister.js]: let's get rid of hyphen and just use "reregister" everywhere

File: src/commands/refocus-collector-reregister.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+const path = `/v1/collectors/${collectorName}/reregister`;
+const url = refocusUrl + path;
+
+// Request to Refocus to re-register collector
+request.post(url)
```
Comment (shriramshankar) [inline src/commands/refocus-collector-reregister.js]: Wondering if this should be a patch instead. For some reason we have also implemented the reregister endpoint on the refocus side as a POST.

File: test/commands/reregister.js [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+  const missingCollectorNameError = 'error: You must specify a collector name\n';
+  const missingUrlError = 'error: You must specify the url of the refocus instance\n';
+  const missingTokenError = 'error: You must specify an access token\n';
+
+  before(() => {
+    nock(refocusUrl, {
+      reqheaders: { authorization: accessToken },
+    })
+    .post('/v1/collectors/collector1/reregister')
+    .reply(201, { collectorToken });
```
Comment (shriramshankar) [inline test/commands/reregister.js]: I think this should be 200. Also, using the constants from  `/src/constants.js` here and in line 40 , will make it clear that these are http status codes.

File: src/commands/index.js (hunk 1 above)
Comment (harshkothari410) [reply src/commands/index.js]: Ok

File: src/commands/refocus-collector-reregister.js (hunk 4 above)
Comment (harshkothari410) [reply src/commands/refocus-collector-reregister.js]: So should I use patch or post ? @shriramshankar

Comment (shriramshankar) [reply src/commands/refocus-collector-reregister.js]: @iamigo Any thoughts on this?

Comment (iamigo) [reply src/commands/refocus-collector-reregister.js]: all these commands are POST requests

**Comment by harshkothari410** [comment]
@iamigo @shriramshankar updated PR.

File: src/commands/refocus-collector-reregister.js [hunk 6]
```diff
@@ -37,15 +37,15 @@ if (!collectorName) {
...
+console.log('Reregister =>', collectorName, refocusUrl, accessToken);
 const path = `/v1/collectors/${collectorName}/reregister`;
 const url = refocusUrl + path;
 
 // Request to Refocus to re-register collector
 request.post(url)
   .set('Authorization', accessToken)
 .then(() => {
-  logger.info(`Collecter ${collectorName} is successfully re-registered.`);
+  logger.info(`Collecter ${collectorName} is successfully reregistered.`);
```
Comment (iamigo) [inline src/commands/refocus-collector-reregister.js]: `is successfully reregistered.` should be `has been reregistered.`

File: src/commands/index.js [hunk 7]
```diff
@@ -24,24 +24,24 @@ const pj = require('../../package.json');
...
 program
 .version(pj.version)
-.command('register --name <name> --url <url> --token <token>',
-  'Register collector with an instance of Refocus')
 .command('start --collectorName <name> --refocusUrl <url> --accessToken <token>',
   'Start collector')
 .command('stop --name <name>', 'Stop collector')
 .command('status --name <name>', 'Show current status of collector')
-.command('deregister --name <name>', 'Deregister collector');
+.command('deregister --name <name>', 'Deregister collector')
```
Comment (iamigo) [inline src/commands/index.js]: `--collectorName`?

File: src/commands/refocus-collector-start.js [hunk 8]
```diff
@@ -19,7 +19,7 @@ program
   .option('-n, --collectorName <collectorName>', 'The name of the collector to be started')
   .option('-n, --refocusUrl <refocusUrl>', 'The url of the refocus instance this collector' +
     ' will send to')
-.option('-n, --accessToken <accessToken>', 'A valid refocus token')
+  .option('-n, --accessToken <accessToken>', 'A valid refocus token')
```
Comment (iamigo) [inline src/commands/refocus-collector-start.js]: I think you can only have one `-n` option

File: src/commands/refocus-collector-reregister.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+console.log('Reregister =>', collectorName, refocusUrl, accessToken);
+const path = `/v1/collectors/${collectorName}/reregister`;
+const url = refocusUrl + path;
+
+// Request to Refocus to re-register collector
+request.post(url)
+  .set('Authorization', accessToken)
+.then(() => {
+  logger.info(`Collecter ${collectorName} is successfully reregistered.`);
```
Comment (iamigo) [inline src/commands/refocus-collector-reregister.js]: `Collector`

File: src/commands/refocus-collector-start.js (hunk 8 above)
Comment (harshkothari410) [reply src/commands/refocus-collector-start.js]: I did not change this. Just fixed style. @iamigo

File: src/commands/index.js (hunk 7 above)
Comment (harshkothari410) [reply src/commands/index.js]: If I will do this I need to change command files that is not part of story.

Comment (harshkothari410) [reply src/commands/index.js]: So let me know what to do @iamigo

File: src/commands/refocus-collector-start.js (hunk 8 above)
Comment (iamigo) [reply src/commands/refocus-collector-start.js]: Yeah, but can you please update them while you're modifying this file?
use -n for collectorName
use -u for refocusUrl
use -t for accessToken

Comment (harshkothari410) [reply src/commands/refocus-collector-start.js]: Done 👍

**Pull request merged**